I tried mssql-jdbc 12.10.0 on native, and I got this exception:

```
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: Unable to locate specified class: com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.getCurrentClassPath(ConfigurableRetryLogic.java:327)
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.readFromFile(ConfigurableRetryLogic.java:339)
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.setUpStatementRules(ConfigurableRetryLogic.java:207)
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.<init>(ConfigurableRetryLogic.java:89)
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.getInstance(ConfigurableRetryLogic.java:106)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.connectInternal(SQLServerConnection.java:2600)
	at com.microsoft.sqlserver.jdbc.SQLServerConnection.connect(SQLServerConnection.java:2207)
	at com.microsoft.sqlserver.jdbc.SQLServerDriver.connect(SQLServerDriver.java:1320)
	at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:225)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:585)
	at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:566)
	at java.base@24.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
	at java.base@24.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1093)
	at java.base@24.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
	... 4 more
Caused by: java.lang.ClassNotFoundException: com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:215)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.hub.ClassForNameSupport.forName(ClassForNameSupport.java:183)
	at java.base@24.0.1/java.lang.Class.forName(DynamicHub.java:1509)
	at java.base@24.0.1/java.lang.Class.forName(DynamicHub.java:1468)
	at java.base@24.0.1/java.lang.Class.forName(DynamicHub.java:1461)
	at com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic.getCurrentClassPath(ConfigurableRetryLogic.java:294)
	... 18 more

```

https://github.com/microsoft/mssql-jdbc/blob/a0ea23a0de0ec314b9836664cd158c4a795174dc/src/main/java/com/microsoft/sqlserver/jdbc/ConfigurableRetryLogic.java#L294

I will update [MsSQLReflections](https://github.com/quarkusio/quarkus/blob/main/extensions/jdbc/jdbc-mssql/deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java) accordingly.

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -122,7 +122,7 @@
         <postgresql-jdbc.version>42.7.5</postgresql-jdbc.version>
         <mariadb-jdbc.version>3.5.3</mariadb-jdbc.version>
         <mysql-jdbc.version>8.3.0</mysql-jdbc.version>
-        <mssql-jdbc.version>12.8.1.jre11</mssql-jdbc.version>
+        <mssql-jdbc.version>12.10.0.jre11</mssql-jdbc.version>
         <adal4j.version>1.6.7</adal4j.version>
         <oracle-jdbc.version>23.6.0.24.10</oracle-jdbc.version>
         <derby-jdbc.version>10.16.1.1</derby-jdbc.version>

--- a/extensions/jdbc/jdbc-mssql/deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java
+++ b/extensions/jdbc/jdbc-mssql/deployment/src/main/java/io/quarkus/jdbc/mssql/deployment/MsSQLReflections.java
@@ -19,6 +19,10 @@ public final class MsSQLReflections {
         //We register it for the sake of people not using Agroal.
         final String driverName = "com.microsoft.sqlserver.jdbc.SQLServerDriver";
         reflectiveClass.produce(ReflectiveClassBuildItem.builder(driverName).build());
+
+        // https://github.com/quarkusio/quarkus/pull/48082#issuecomment-2912550391
+        String className = "com.microsoft.sqlserver.jdbc.ConfigurableRetryLogic";
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(className).build());
     }
 
 }

